src/task.ts

```diff
@@ -125,6 +125,7 @@ export abstract class TaskBase {
      */
     constructor(public id: TaskID, protected action: BackingAction) {
         this.state = TaskState.Running;
+        this.isPlayed = false;
```

Comment:
This checking: do we need to make sure this is updated for non-compound tasks too, or is that already handled?

Reply:
Well this is being updated in the `TaskBase` class, so this change should apply to compound and non-compound tasks alike. Does that answer your question or did I misunderstand it?

Reply:
Hmm so I get that here we are setting the default initial value to be `false`, which will apply to both compound and non-compound tasks. However, in addition to that, this PR also provides a way for compound tasks to update that default value through `trySetIsPlayed()`. I'm wondering if we need an equivalent for non-compound tasks too?

Reply:
Ah, I found where this is set for atomic (i.e not compound) tasks.

Essentially, as we iterate over History events, we set the `IsPlayed` flag of those atomic events, because atomic results are stored in individual History events. See [here](https://github.com/Azure/azure-functions-durable-js/blob/f716f842401ccfdeaf00c989a317944116110627/src/taskorchestrationexecutor.ts#L357-L360)

Reply:
Makes sense, thank you for looking into it!